internal/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,254 @@
+	})
+}
+
+// it returns generic http filter needed for applying local rate limit
+func localHttpFilterPatch() *envoyfilter.ConfigPatch {
+	return &v1alpha3.EnvoyFilter_EnvoyConfigObjectPatch{
+		ApplyTo: v1alpha3.EnvoyFilter_HTTP_FILTER,
+		Match: &v1alpha3.EnvoyFilter_EnvoyConfigObjectMatch{
+			Context: v1alpha3.EnvoyFilter_SIDECAR_INBOUND,
+			ObjectTypes: &v1alpha3.EnvoyFilter_EnvoyConfigObjectMatch_Listener{
+				Listener: &v1alpha3.EnvoyFilter_ListenerMatch{
+					FilterChain: &v1alpha3.EnvoyFilter_ListenerMatch_FilterChainMatch{
+						Filter: &v1alpha3.EnvoyFilter_ListenerMatch_FilterMatch{
+							Name: "envoy.filters.network.http_connection_manager",
+						},
+					},
+				},
+			},
+		},
+		Patch: &v1alpha3.EnvoyFilter_Patch{
+			Operation: v1alpha3.EnvoyFilter_Patch_INSERT_BEFORE,
+			Value: &structpb.Struct{Fields: map[string]*structpb.Value{
+				"name": structpb.NewStringValue(LocalRateLimitFilterName),
+				"typed_config": structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+					"@type":    structpb.NewStringValue(TypedStruct),
+					"type_url": structpb.NewStringValue(LocalRateLimitFilterUrl),
+					"value": structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"stat_prefix": structpb.NewStringValue("http_local_rate_limiter"),
+					}}),
+				}}),
+			}}},
+	}
+}
+
+// RateLimitConfigPatch generates Istio-compatible ConfigPatch containing local rate limit configuration
+func (rl *RateLimit) RateLimitConfigPatch() *envoyfilter.ConfigPatch {
+	return &envoyfilter.ConfigPatch{
+		ApplyTo: v1alpha3.EnvoyFilter_HTTP_ROUTE,
+		Match: &v1alpha3.EnvoyFilter_EnvoyConfigObjectMatch{
+			Context: v1alpha3.EnvoyFilter_SIDECAR_INBOUND,
+		},
+		Patch: &v1alpha3.EnvoyFilter_Patch{
+			Operation: v1alpha3.EnvoyFilter_Patch_MERGE,
+			Value: &structpb.Struct{Fields: map[string]*structpb.Value{
+				"route": structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+					"rate_limits": structpb.NewListValue(&structpb.ListValue{Values: []*structpb.Value{
+						structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+							"actions": func() *structpb.Value {
+								var actVal []*structpb.Value
+								for _, a := range rl.actions {
+									actVal = append(actVal, a.Value())
+								}
+								return structpb.NewListValue(&structpb.ListValue{Values: actVal})
+							}(),
+						}}),
+					}}),
+				}}),
+				"typed_per_filter_config": structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+					LocalRateLimitFilterName: structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+						"@type":    structpb.NewStringValue(rl.limitType),
+						"type_url": structpb.NewStringValue(rl.limityTypeUrl),
+						"value": structpb.NewStructValue(&structpb.Struct{Fields: map[string]*structpb.Value{
+							"stat_prefix":                structpb.NewStringValue("rate_limit"),
```

Comment:
I don't think it's necessary right now. There will be no changes, and I'd rather keep these fields hardcoded rather than moving them to const values. 

it's a good candidate for making entire `value` as config for rate limit. But unless we plan implementing changing things, like filter weights for rate limits, I see no real benefit from it now.